### PR TITLE
EQL: code cleanup and further tests

### PR DESCRIPTION
**Pull request by astefan**
* adds Pipe tests to all function
* missing Processor tests for `string` and `substring` functions
* use an overall `input` named parameter to avoid confusion with the already existent `source`, `src` in all functions related classes
* other minor bug-fixes and code cleanups

A wide NodeSubclass test class to be added later with https://github.com/elastic/elasticsearch/issues/53862.

Addresses #54568.

### CODE REVIEW

File: x-pack/plugin/ql/src/test/java/org/elasticsearch/xpack/ql/tree/NodeSubclassTests.java [hunk 1]
```diff
@@ -559,7 +559,7 @@ private Object makeMap(Class<? extends Node<?>> toBuildClass, ParameterizedType
 
     private int randomSizeForCollection(Class<? extends Node<?>> toBuildClass) {
         int minCollectionLength = 0;
-        int maxCollectionLength = 10;
+        int maxCollectionLength = 3;
```
Comment (matriv) [inline x-pack/plugin/ql/src/test/java/org/elasticsearch/xpack/ql/tree/NodeSubclassTests.java:562]: Why did you reduce this?

**Review by matriv — APPROVED** [review]
LGTM. Nice work adding all those missing tests!
Left a couple of minor questions.

File: x-pack/plugin/ql/src/test/java/org/elasticsearch/xpack/ql/tree/NodeSubclassTests.java (hunk 1 above)
Comment (astefan) [reply x-pack/plugin/ql/src/test/java/org/elasticsearch/xpack/ql/tree/NodeSubclassTests.java:562]: Good catch. That's a leftover from an attempt to add EqlNodeSubclassTests to EQL. Will revert it.

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/ConcatFunctionPipe.java [hunk 2]
```diff
@@ -85,6 +85,10 @@ public ConcatFunctionProcessor asProcessor() {
         return new ConcatFunctionProcessor(processors);
     }
 
+    public List<Pipe> values() {
```
[the remark this answers is not in the document]
Comment (astefan) [reply x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/expression/function/scalar/string/ConcatFunctionPipe.java:88]: Yes, can be package private.

**Review by costin — APPROVED** [review]
LGTM!

**Pull request merged**